awx/main/tests/functional/api/test_inventory.py

```diff
@@ -516,6 +515,49 @@ def test_vault_credential_not_allowed_via_related(
         assert 'Credentials of type insights and vault' in r.data['msg']
         assert 'disallowed for scm inventory sources' in r.data['msg']
 
+    def test_view_multiple_credentials(self, project, inventory, get, admin_user):
+        inv_src = InventorySource.objects.create(
+            inventory=inventory, name='foobar', source='scm',
+            source_project=project, source_path=''
+        )
+        aws_ct = CredentialType.defaults['aws']()
+        aws_ct.save()
+        aws_cred = Credential.objects.create(name='is_aws', credential_type=aws_ct)
+        gce_ct = CredentialType.defaults['gce']()
+        gce_ct.save()
+        gce_cred = Credential.objects.create(name='is_gce', credential_type=gce_ct)
+        inv_src.credentials.add(aws_cred, gce_cred)
+        r = get(
+            url=reverse('api:inventory_source_credentials_list', kwargs={'pk': inv_src.pk}),
+            user=admin_user, expect=200
+        )
+        assert r.data['count'] == 2
+        r = get(url=inv_src.get_absolute_url(), user=admin_user, expect=200)
+        # the credential field could be either of the credentials
+        assert r.data['credential'] in [gce_cred.pk, aws_cred.pk]
+
+    def test_associate_multiple_credentials(self, project, inventory, post, patch, admin_user):
+        inv_src = InventorySource.objects.create(
+            inventory=inventory, name='foobar', source='scm',
+            source_project=project, source_path=''
+        )
+        aws_ct = CredentialType.defaults['aws']()
+        aws_ct.save()
+        aws_cred = Credential.objects.create(name='is_aws', credential_type=aws_ct)
+        gce_ct = CredentialType.defaults['gce']()
+        gce_ct.save()
+        gce_cred = Credential.objects.create(name='is_gce', credential_type=gce_ct)
+        for cred in [aws_cred, gce_cred]:
+            post(
+                url=reverse('api:inventory_source_credentials_list', kwargs={'pk': inv_src.pk}),
+                data={'id': cred.pk},
+                user=admin_user,
+                expect=204
+            )
+        assert set(inv_src.credentials.all()) == set([aws_cred, gce_cred])
+        patch(url=inv_src.get_absolute_url(), data={'credential': gce_cred.pk}, user=admin_user, expect=200)
+        assert set(inv_src.credentials.all()) == set([gce_cred])
```

Comment:
@ryanpetrello this is where expectations get really murky.

If you PATCH `{"credential": 3}` does that mean "replace the entire credential list with just credential 3"?

I _think_ this is the way it works for job templates. If not, this should be changed.

Reply:
So this is probably a little tricky because it's not _exactly_ analogous to JT.  At least with JT, you had rules like "if you specify a `.credential`, set _that_ as the only machine credential.

```
If you PATCH {"credential": 3} does that mean "replace the entire credential list with just credential 3"?
```

...maybe?  This is probably the _closest_ behavior to what JT v1 assignment does.